samtools_mpileup_tool/multi_samtools_mpileup.py

```diff
@@ -0,0 +1,246 @@
+            interval = "{}:{}-{}".format(chrom, int(start) + 1, end)
+            yield interval
+
+
+def get_file_size(filename: pathlib.Path) -> int:
+    """ Gets file size """
+    return filename.stat().st_size
+
+
+def yield_formatted_commands(
+    reference_path: str,
+    min_mq: str,
+    normal_bam: str,
+    tumor_bam: str,
+    interval_bed_path: str,
+) -> Generator[str, None, None]:
+    """Yield commands for each BED interval."""
+    for region in yield_bed_regions(interval_bed_path):
+        cmd = CMD_STR.format(
+            reference_path=reference_path,
+            min_mq=min_mq,
+            region=region,
+            normal_bam=normal_bam,
+            tumor_bam=tumor_bam,
+            output=region.replace(":", "-"),
+        )
+        yield cmd
+
+
+def setup_parser():
+    """
+    Loads the parser.
+    """
+    parser = argparse.ArgumentParser(
+        "Internal multithreading samtools mpileup calling."
+    )
+    parser.add_argument("--version", action="version", version=__version__)
+    # Required flags.
+    parser.add_argument("-f", "--reference_path", required=True, help="Reference path.")
+    parser.add_argument(
+        "-r", "--interval_bed_path", required=True, help="Interval bed file."
+    )
+    parser.add_argument("-t", "--tumor_bam", required=True, help="Tumor bam file.")
+    parser.add_argument("-n", "--normal_bam", required=True, help="Normal bam file.")
+    parser.add_argument(
+        "-c", "--thread_count", type=int, required=True, help="Number of thread."
+    )
+    parser.add_argument("-m", "--min_mq", type=int, required=True, help="min MQ.")
+    return parser
+
+
+def process_argv(argv: Optional[List] = None) -> namedtuple:
+    """Processes argv into namedtuple."""
+
+    parser = setup_parser()
+
+    if argv:
+        args, unknown_args = parser.parse_known_args(argv)
+    else:
+        args, unknown_args = parser.parse_known_args()
+
+    args_dict = vars(args)
+    args_dict['extras'] = unknown_args
+    run_args = namedtuple('RunArgs', list(args_dict.keys()))
+    return run_args(**args_dict)
+
+
+def run(run_args):
+    """Main script logic.
+    Creates commands for each BED region and executes in multiple threads.
+    """
+
+    run_commands = list(
+        yield_formatted_commands(
+            run_args.reference_path,
+            run_args.min_mq,
+            run_args.normal_bam,
+            run_args.tumor_bam,
+            run_args.interval_bed_path,
+        )
+    )
+    # Start Queue
+    exceptions = tpe_submit_commands(run_commands, run_args.thread_count)
+    if exceptions:
+        for e in exceptions:
+            logger.error(e)
+        raise ValueError("Exceptions raised during processing.")
+
+    # Check and merge outputs
+    p = pathlib.Path('.')
+    outputs = list(p.glob("*.mpileup"))
+
+    # Sanity check
+    if len(run_commands) != len(outputs):
+        logger.error("Number of output files not expected")
+
```

Comment:
Is it necessary to merge the output files here?

Reply:
no. the varscan/somaticsniper will take a list of mpileup as inputs. so just need to check the number of outputs. 👍